app/views/admin/members/show.html.haml

```diff
@@ -253,6 +253,31 @@
                       %button.btn.btn-default.destroy
                         %i.fa.fa-fw.fa-trash
 
+        .card
+          .card-header
+            = I18n.t('admin.sac.title')
+            = link_to I18n.t("admin.sac.download"), member_sac_path(@member.id)
+          %table.table.table-striped#sac
+            %thead
+              %tr
+                %th= I18n.t('admin.activities.single')
+                %th= I18n.t('admin.sac.category')
+                %th= I18n.t('admin.sac.title')
+            %tbody
+              - total = 0
+              - @activities.select{ |ac| ac.sac_category? or ac.participants.where(:member => @member).first.sac_points?}.each do |activity|
```

Comment:
Can you move this logic to the controller?
It is unavoidable to have a loop here, but a query doesn't really mix with well with a view in my opinion.